client/a8c-for-agencies/sections/sites/site-actions/use-site-actions.ts

```diff
@@ -202,3 +205,226 @@ export default function useSiteActions( {
+					page( `https://wordpress.com/activity-log/${ getBlogId( items[ 0 ] ) }` );
+					dispatch( recordTracksEvent( getActionEventName( 'view_activity', isLargeScreen ) ) );
+				},
+			},
+			{
+				id: 'clone_site_wpcom',
+				label: translate( 'Copy this site' ),
+				icon: external,
+				isEligible( item: SiteData ) {
+					return isWPComSite( item ) && hasBackup( item ) && ! isUrlOnly( item );
+				},
+				callback( items: SiteData[] ) {
+					page( `https://wordpress.com/backup/${ getSiteSlug( items[ 0 ] ) }/clone` );
+					dispatch( recordTracksEvent( getActionEventName( 'clone_site', isLargeScreen ) ) );
+				},
+			},
+			{
+				id: 'clone_site_not_wpcom',
+				label: translate( 'Copy this site' ),
+				isEligible( item: SiteData ) {
+					return ! isWPComSite( item ) && hasBackup( item ) && ! isUrlOnly( item );
+				},
+				callback( items: SiteData[] ) {
+					setDataViewsState( ( prevState: DataViewsState ) => ( {
+						...prevState,
+						selectedItem: items[ 0 ].site.value,
+						type: DATAVIEWS_LIST,
+					} ) );
+					setSelectedSiteFeature( JETPACK_BACKUP_ID );
+					dispatch( recordTracksEvent( getActionEventName( 'clone_site', isLargeScreen ) ) );
+				},
+			},
+			{
+				id: 'site_settings',
+				label: translate( 'Site settings' ),
+				icon: external,
+				isEligible( item: SiteData ) {
+					return isWPComSite( item ) && ! isUrlOnly( item );
+				},
+				callback( items: SiteData[] ) {
+					page( `https://wordpress.com/settings/general/${ getBlogId( items[ 0 ] ) }` );
+					dispatch( recordTracksEvent( getActionEventName( 'site_settings', isLargeScreen ) ) );
+				},
+			},
+			{
+				id: 'view_site',
+				label: translate( 'View site' ),
+				icon: external,
+				isEligible() {
+					return true;
+				},
+				callback( items: SiteData[] ) {
+					page( getUrlWithScheme( items[ 0 ] ) );
+					dispatch( recordTracksEvent( getActionEventName( 'view_site', isLargeScreen ) ) );
+				},
+			},
+			{
+				id: 'visit_wp_admin',
+				label: translate( 'Visit WP Admin' ),
+				icon: external,
+				isEligible( item: SiteData ) {
+					return ! isUrlOnly( item );
+				},
+				callback( items: SiteData[] ) {
+					page( `${ getUrlWithScheme( items[ 0 ] ) }/wp-admin` );
+					dispatch( recordTracksEvent( getActionEventName( 'visit_wp_admin', isLargeScreen ) ) );
+				},
+			},
+			{
+				id: 'remove_site',
+				label: translate( 'Remove site' ),
+				icon: trash,
+				isEligible( item: SiteData ) {
+					return ! isDevSite( item ) && hasRemoveManagedSitesCapability();
+				},
+				RenderModal: createRemoveSiteActionModal( onRefetchSite ),
+				isDestructive: true,
+			},
+			{
+				id: 'delete_site',
+				label: translate( 'Delete site' ),
+				isEligible() {
+					return false; // Feature is always disabled, see canDelete above.
+				},
+				callback() {
+					dispatch( recordTracksEvent( getActionEventName( 'delete_site', isLargeScreen ) ) );
+					// TODO: implement modal. See setShowDeleteDevSiteDialog in SiteActions.
```

Comment:
Even if it's not used at the moment, I think it'd be best to implement this modal following what [DevSiteDeleteConfirmationDialog](https://github.com/Automattic/wp-calypso/blob/add/actions-to-a4a-sites/client/a8c-for-agencies/sections/sites/dev-site-delete-confirmation-dialog/index.tsx#L20) does.

Reply:
Implemented the basic modal at https://github.com/Automattic/wp-calypso/pull/98276/commits/b74f61f37398ef61b193e89ae07a5cefd01853fa